test/libsolidity/smtCheckerTests/types/enum_range.sol

```diff
@@ -0,0 +1,9 @@
+pragma experimental SMTChecker;
+
+contract C
+{
+	enum D { Left, Right }
+	function f(D a) public pure {
+		assert(a == D.Left || a == D.Right);
+	}
+}
```

Comment:
Can you please add test cases for other types that have range support i.e., integer, fixed bytes? Or are they already there somewhere? Also are both signed and unsigned integer types of different widths covered?

Reply:
Those have tests already. The two things added here were `address` and constraining the range for `enum`s more

Reply:
Okay! Looks good to me if fixed bytes are also constrained and something like `byte x; assert(x < 256);` does not produce a warning.

Reply:
An example like that is disallowed by the TypeChecker of the compiler.
I checked that the constraints are added.
Nonetheless, I added a test that would detect if SMTChecker would not add the constraints, to prevent accidental future change.